### ISSUE TITLE
Error generating results on staging environment

### ISSUE THREAD

**Issue by olafveerman** [WorldBank-Transport/Rural-Road-Accessibility#138]
When running analysis on 'bigger' road networks, the OSRM process on the staging environment fails. The following road network is 148mb:

![screenshot from 2017-05-24 08-22-08](https://cloud.githubusercontent.com/assets/751330/26403721/3972c7a6-405d-11e7-89c2-25b1f7f4404d.png)

- running the same analysis locally succeeds
- analysis with the same inputs, but a smaller road network (38mb) succeeds on the staging environment

This is the log of the analysis container:

```
[12:22:29.551] [main] Max running processes set at 1
[12:23:06.554] [OSRM] Generation started
err Error: /code/scripts/osm2osrm.sh: line 34:    19 Killed                  $OSRM_EXTRACT -p profile.lua road-network.osm
terminate called after throwing an instance of 'osrm::util::exception'
  what():  Error opening road-network.osrm.enw
/code/scripts/osm2osrm.sh: line 35:    23 Aborted                 $OSRM_CONTRACT road-network.osrm

    at /code/app/index.js:231:32
    at ChildProcess.exithandler (child_process.js:211:5)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at maybeClose (internal/child_process.js:886:16)
    at Socket.<anonymous> (internal/child_process.js:342:11)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at Pipe._handle.close [as _onclose] (net.js:497:12)
[12:23:15.924] [error] /code/scripts/osm2osrm.sh: line 34:    19 Killed                  $OSRM_EXTRACT -p profile.lua road-network.osm
terminate called after throwing an instance of 'osrm::util::exception'
  what():  Error opening road-network.osrm.enw
/code/scripts/osm2osrm.sh: line 35:    23 Aborted                 $OSRM_CONTRACT road-network.osrm

[12:23:15.924] [error] Error: /code/scripts/osm2osrm.sh: line 34:    19 Killed                  $OSRM_EXTRACT -p profile.lua road-network.osm
terminate called after throwing an instance of 'osrm::util::exception'
  what():  Error opening road-network.osrm.enw
/code/scripts/osm2osrm.sh: line 35:    23 Aborted                 $OSRM_CONTRACT road-network.osrm

    at /code/app/index.js:231:32
    at ChildProcess.exithandler (child_process.js:211:5)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at maybeClose (internal/child_process.js:886:16)
    at Socket.<anonymous> (internal/child_process.js:342:11)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at Pipe._handle.close [as _onclose] (net.js:497:12)
```

cc @xijielu @danielfdsilva

**Comment by olafveerman** [comment]
Did some more investigation, but haven't reached a conclusion yet. Here are some notes.

- the OSM -> OSRM conversion [happens here](https://github.com/WorldBank-Transport/rra-datapipeline/blob/develop/rra-analysis/scripts/osm2osrm.sh#L34). When dealing with a big road network on Hyper, the process doesn't generate all the files it typically does.  
  ```
  $ hyper exec [containerId] ls -l /conversion/p1s1
  lib
  profile.lua
  road-network.osm
  road-network.osrm
  road-network.osrm.timestamp
  stxxl.errlog
  stxxl.log
  ```
- sometimes the process gets to generate the `road-network.osrm`, other times not even that
- I tried to replicate it locally with Docker, limiting the container to 1cpu and 512mb ram. While it was slow, the process did generate the results

Next step: spin up a more powerful Hyper container to rule out cpu/ram limitations as the cause.

cc @danielfdsilva

**Comment by olafveerman** [comment]
This was solved by increasing the size of the Hyper container that the analysis runs on. We'll add a note to the documentation about this. For future reference:

- S4 - the default Hyper container - has been tested and works with road networks until at least 38mb
- L1 - 4 cpu & 4GB mem - works well for the bigger road networks we tested with (148mb)

See the [Hyper docs](https://hyper.sh/pricing.html) for more info about the containers.
